// @flowglad/node/resources/subscriptions.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.Subscriptions = void 0;
const resource_1 = require("../core/resource.js");
const path_1 = require("../internal/utils/path.js");
class Subscriptions extends resource_1.APIResource {
    /**
     * Create Subscription
     */
    create(body, options) {
        return this._client.post('/api/v1/subscriptions', { body, ...options });
    }
    /**
     * Get Subscription
     */
    retrieve(id, options) {
        return this._client.get((0, path_1.path) `/api/v1/subscriptions/${id}`, options);
    }
    /**
     * List Subscriptions
     */
    list(query = {}, options) {
        return this._client.get('/api/v1/subscriptions', { query, ...options });
    }
    /**
     * Adjust a Subscription
     */
    adjust(id, body, options) {
        return this._client.post((0, path_1.path) `/api/v1/subscriptions/${id}/adjust`, { body, ...options });
    }
    /**
     * Cancel a Subscription
     */
    cancel(id, body, options) {
        return this._client.post((0, path_1.path) `/api/v1/subscriptions/${id}/cancel`, { body, ...options });
    }
}
exports.Subscriptions = Subscriptions;
