// @flowglad/node/resources/subscriptions.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../core/resource.mjs";
import { path } from "../internal/utils/path.mjs";
export class Subscriptions extends APIResource {
    /**
     * Create Subscription
     */
    create(body, options) {
        return this._client.post('/api/v1/subscriptions', { body, ...options });
    }
    /**
     * Get Subscription
     */
    retrieve(id, options) {
        return this._client.get(path `/api/v1/subscriptions/${id}`, options);
    }
    /**
     * List Subscriptions
     */
    list(query = {}, options) {
        return this._client.get('/api/v1/subscriptions', { query, ...options });
    }
    /**
     * Adjust a Subscription
     */
    adjust(id, body, options) {
        return this._client.post(path `/api/v1/subscriptions/${id}/adjust`, { body, ...options });
    }
    /**
     * Cancel a Subscription
     */
    cancel(id, body, options) {
        return this._client.post(path `/api/v1/subscriptions/${id}/cancel`, { body, ...options });
    }
}
