// @flowglad/shared/dist/cjs/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var index_exports = {};
__export(index_exports, {
  FlowgladActionKey: () => import_types.FlowgladActionKey,
  HTTPMethod: () => import_types.HTTPMethod,
  cancelSubscriptionSchema: () => import_actions.cancelSubscriptionSchema,
  constructCheckFeatureAccess: () => import_utils.constructCheckFeatureAccess,
  constructCheckUsageBalance: () => import_utils.constructCheckUsageBalance,
  constructGetPrice: () => import_utils.constructGetPrice,
  constructGetProduct: () => import_utils.constructGetProduct,
  createAddPaymentMethodCheckoutSessionSchema: () => import_actions.createAddPaymentMethodCheckoutSessionSchema,
  createCheckoutSessionSchema: () => import_actions.createCheckoutSessionSchema,
  createProductCheckoutSessionSchema: () => import_actions.createProductCheckoutSessionSchema,
  createSubscriptionSchema: () => import_actions.createSubscriptionSchema,
  createUsageEventSchema: () => import_actions.createUsageEventSchema,
  flowgladActionValidators: () => import_actions.flowgladActionValidators,
  getBaseURL: () => import_utils.getBaseURL,
  updateCustomerSchema: () => import_actions.updateCustomerSchema
});
module.exports = __toCommonJS(index_exports);
var import_types = require("./types");
var import_actions = require("./actions");
var import_utils = require("./utils");
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  FlowgladActionKey,
  HTTPMethod,
  cancelSubscriptionSchema,
  constructCheckFeatureAccess,
  constructCheckUsageBalance,
  constructGetPrice,
  constructGetProduct,
  createAddPaymentMethodCheckoutSessionSchema,
  createCheckoutSessionSchema,
  createProductCheckoutSessionSchema,
  createSubscriptionSchema,
  createUsageEventSchema,
  flowgladActionValidators,
  getBaseURL,
  updateCustomerSchema
});
